components/raftstore/src/store/worker/read.rs

```diff
@@ -1546,135 +1627,274 @@ mod tests {
             meta.readers.insert(2, read_delegate);
         }
 
-        let mut read_id = ThreadReadId::new();
-        let mut snap_cache = Box::new(None);
-
-        let read_id_copy = Some(read_id.clone());
-
-        let mut read_context = Some(LocalReadContext {
-            read_id: &mut read_id,
-            snap_cache: &mut snap_cache,
-        });
-
-        let (_, delegate) = store_meta.get_executor_and_len(1);
+        let (len, delegate) = store_meta.get_executor_and_len(1);
+        assert_eq!(2, len);
         let mut delegate = delegate.unwrap();
+        assert_eq!(1, delegate.region.id);
         let tablet = delegate.get_tablet();
         assert_eq!(kv_engine.as_inner().path(), tablet.as_inner().path());
-        let snapshot = delegate.get_snapshot(read_id_copy.clone(), &mut read_context);
-        assert_eq!(
-            b"val1".to_vec(),
-            *snapshot.get_value(b"a1").unwrap().unwrap()
-        );
 
-        let (_, delegate) = store_meta.get_executor_and_len(2);
+        let (len, delegate) = store_meta.get_executor_and_len(2);
+        assert_eq!(2, len);
         let mut delegate = delegate.unwrap();
+        assert_eq!(2, delegate.region.id);
         let tablet = delegate.get_tablet();
         assert_eq!(kv_engine.as_inner().path(), tablet.as_inner().path());
-        let snapshot = delegate.get_snapshot(read_id_copy, &mut read_context);
+    }
+
+    #[test]
+    fn test_snap_across_regions() {
+        let store_id = 2;
+        let store_meta = Arc::new(Mutex::new(StoreMeta::new(0)));
+        let (_tmp, mut reader, rx) = new_reader("test-local-reader", store_id, store_meta.clone());
+
+        fn prepare_read_delegate(
```

Comment:
Bad style, should avoid nested function.

Reply:
Done